src/ArduinoGraphics.cpp

```diff
@@ -124,6 +124,51 @@ void ArduinoGraphics::noStroke()
   _stroke = false;
 }
 
+void ArduinoGraphics::ellipse(int x, int y, int width, int height)
+{
+  if (!_stroke && !_fill) {
+    return;
+  }
+
+  float theta;
+  int r1 = width / 2;
+  int r2 = height / 2;
+  for (int angle = 0; angle < 360; angle += 1) {
+    theta = angle * 3.14 / 180;
+    int xi = r1 * cos(theta);
+    int yi = r2 * sin(theta);
+    if (_stroke) {
+        // stroke
+        set(x + xi, y - yi, _strokeR, _strokeG, _strokeB);
+      } else if (_fill) {
+        // fill
+        set(x + xi, y - yi, _fillR, _fillG, _fillB);
+      }
+  }
+}
+
+void ArduinoGraphics::circle(int x, int y, int radius)
+{
+  //ellipse(x, y, 2 * radius, 2 * radius);
```

Comment:
Why did you decide not to functionalize the shared code? I see from this comment that you started out to do that at one point.

Reply:
Just for reducing the number of function calling.

Reply:
I'll leave that choice up to you and whoever does the final review on this, but I would request that you remove the commented out code, since it doesn't serve any purpose.